src/esm_runscripts/prepare.py

```diff
@@ -729,3 +730,60 @@ def set_logfile(config):
         "experiment_log_file", log_file_path
     )
     return config
+
+
+def check_config_for_warnings_errors(config):
+
+    triggers = {"error": {"note_function": esm_parser.user_error}}
+
+    # Find conditions to warn (avoid warning more than once)
+    last_jobtype = config["general"].get("last_jobtype", "")
+    isresubmitted = last_jobtype == config["general"]["jobtype"]
+    isinteractive = config["general"].get("isinteractive", "")
+
+    # Only warn if it is an interactive session or while submitted
+    if not isresubmitted or isinteractive:
+        triggers["warning"] = {"note_function": esm_parser.user_note}
+
+    for trigger, trigger_info in triggers.items():
+        warn_error(config, trigger, trigger_info["note_function"])
+
+    return config
+
+def warn_error(config, trigger, note_function):
```

Comment:
It would be nice to include in the docstring a minimal example of what would be written in the config, and what the terminal output would look like.